### PR TITLE
Add a grouping overlay

### PR DESCRIPTION
**Pull request by phschaad**
Support visualization of logical SDFG groups

grouping_demo

### CODE REVIEW

File: src/renderer/renderer.ts [hunk 1]
```diff
@@ -2453,7 +2498,7 @@ export class SDFGRenderer {
         if (this.external_mouse_handler) {
             const ext_mh_dirty = this.external_mouse_handler(
                 evtype, event, { x: mouse_x, y: mouse_y }, elements,
-                this, this.selected_elements, ends_drag, this.sdfv_instance
+                this, this.selected_elements, this.sdfv_instance
```
Comment (tbennun) [inline src/renderer/renderer.ts:2501]: What's this sneaky line doing here?

**Review by tbennun — CHANGES_REQUESTED** [review]
I think the overlay should be turned on by default. Thoughts?

File: src/renderer/renderer.ts (hunk 1 above)
Comment (phschaad) [reply src/renderer/renderer.ts:2501]: This is in preparation for being able to create groupings from VSCode via box select. When multiple elements are selected via Ctrl.+Click, the information panel stops showing details from a specific element and instead displays a message saying 'multiple elements selected'. However, this `ends_drag` check was blocking this from happening, if the selection was performed via box select - which makes no sense. I am not sure why this check was there to begin with, but testing didn't bring up any issues with respect to dragging and releasing that would hint to this check being required..

**Comment by phschaad** [comment]
Agree that the overlay should be enabled by default - didn't think of already doing that, thanks..! :-)

File: src/renderer/renderer.ts (hunk 1 above)
Comment (tbennun) [reply src/renderer/renderer.ts:2504]: Nice!

**Review by tbennun — APPROVED** [review]
LGTM

**Pull request merged**